scylla_phot_culls.py

```diff
@@ -0,0 +1,183 @@
+# coding: utf-8
+"""
+Make quality cuts on the Scylla photometry, save new catalog, and optionally plot data spatially and in a CMD.
+Petia YMJ, Mar 2021
+"""
+import argparse
+import os
+import time
+import numpy as np
+from astropy.table import Table
+import matplotlib.pyplot as plt
+from scipy.stats import gaussian_kde
+plt.ion()
+
+
+def make_cuts(catalog_in='data/smc_6-15891_SMC-3956ne-9632/proc_default/15891_SMC-3956ne-9632.st.fits',
+              catalog_out=None,
+              filters=['F475W', 'F814W'],
+              crowdcut=[0.3],
+              sharpcut=[0.15],
+              roundcut=[0.6],
+              errcut=None,
+              plot=False):
+    """
+    Make quality cuts on the input photometry catalog file and record them in a new catalog file.
+
+    Parameters
+    ----------
+    catalog_in : str
+        catalog file to which to apply photometry quality cuts
+    catalog_out : str
+        catalog file to which to record new photometry
+    filters : str or a list of strings
+        HST filters to which to apply quality cuts. Example: ['F336W','F475W','F814W']
+    crowdcut : float or a list of floats
+        If single value, cut is applied to all filters. If multiple values, the number of values has to match the
+        number of filters, and the cuts are applied in order based on the filter order.
+    sharpcut : float or a list of floats
+        If single value, cut is applied to all filters. If multiple values, the number of values has to match the
+        number of filters, and the cuts are applied in order based on the filter order.
+    roundcut : float or a list of floats
+        If single value, cut is applied to all filters. If multiple values, the number of values has to match the
+        number of filters, and the cuts are applied in order based on the filter order.
+    errcut : float or a list of floats
+        If single value, cut is applied to all filters. If multiple values, the number of values has to match the
+        number of filters, and the cuts are applied in order based on the filter order.
+    plot : Boolean
+        If True, generate two plots as a check and a visualization of the data with quality cuts:
+        1) spatial plot
+        2) CMDs: if more than one filter is provided, plot a CMD with color=filter1-filter2 and mag=filter1
+    """
+
+    field_id = catalog_in.split('.st')[0].split('/15891_')[1]
```

Comment:
Change 
`field_id = catalog_in.split('.st')[0].split('/15891_')[1]` to 
`field_id = catalog_in.split('.st')[0].split('/15891_')[-1]` 
so that if any of the folders leading to the files contain "/15891_" then there isn't any issues.

Reply:
It turns out that both of these work, so I kept the code as is.